File: csharp/cars-assemble/CarsAssemble.cs
using System;
using System.Reflection.Metadata;

static class AssemblyLine
{

    public static double SuccessRate(int speed)
    {
        if (speed <= 0) { 
            return 0.0;

        } else if (speed <= 4) { 
            return 1.0;

        } else if (speed <= 8) {
            return 0.90;

        } else if (speed == 9) {
            return 0.80;

        } else if (speed == 10) {
            return 0.77;

        } else {
            throw new ArgumentOutOfRangeException(speed.ToString());
        }
    }

    public static double ProductionRatePerHour(int speed)
    {
        const int carsPerHourAtSpeed1 = 221;

        return carsPerHourAtSpeed1 * speed * SuccessRate( speed );
    }

    public static int WorkingItemsPerMinute(int speed)
    {   
        const double minutesInHour = 60;

        return (int)(ProductionRatePerHour( speed ) / minutesInHour);

    }
}

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {

    public double productionRatePerHour(int speed) {
        final double carsAtOneSpeed = 221;

        double carsAtAtualSpeed = carsAtOneSpeed * speed;
        double successRate;

        if (1 <= speed && speed <= 4) {
            successRate = 1;
        } else if (5 <= speed && speed <= 8) {
            successRate = 0.9;
        } else if (speed == 9) {
            successRate = 0.8;
        } else if (speed == 10) {
            successRate = 0.77;
        } else {
            return 0;
        }

        return carsAtAtualSpeed * successRate;
    }

    public int workingItemsPerMinute(int speed) {
        final double minutesInHour = 60;

        return (int)(this.productionRatePerHour(speed)/minutesInHour);
    }
}
